### PR TITLE
feat: Integrate Vault backend

### LINKED ISSUE #433: jhipster-registry not starting when using vault as config backend
**Overview of the issue**

Adding vault as a config server backend with config.server.bootstrap=true makes the registry fail on start.

**Reproduce the error**
In a JHipster microservices architecture with docker deployment, add the following environment vars to jhipster-registry.yml:
```yml
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_TYPE=native
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_SEARCH_LOCATIONS=file:./central-config
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_TYPE=vault
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_HOST=localhost
      - SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_PORT=8200
```
Exception:
```
jhipster-registry_1   | java.lang.IllegalStateException: No HttpServletRequest available
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.HttpRequestConfigTokenProvider.getToken(HttpRequestConfigTokenProvider.java:41)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.VaultEnvironmentRepository.getToken(VaultEnvironmentRepository.java:118)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.VaultEnvironmentRepository.read(VaultEnvironmentRepository.java:109)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.AbstractVaultEnvironmentRepository.findOne(AbstractVaultEnvironmentRepository.java:88)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.EnvironmentRepository.findOne(EnvironmentRepository.java:31)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.CompositeEnvironmentRepository.findOne(CompositeEnvironmentRepository.java:66)
jhipster-registry_1   | 	at org.springframework.cloud.config.server.environment.EnvironmentRepositoryPropertySourceLocator.locate(EnvironmentRepositoryPropertySourceLocator.java:55)
jhipster-registry_1   | 	at org.springframework.cloud.bootstrap.config.PropertySourceLocator.locateCollection(PropertySourceLocator.java:52)
jhipster-registry_1   | 	at org.springframework.cloud.bootstrap.config.PropertySourceLocator.locateCollection(PropertySourceLocator.java:47)
jhipster-registry_1   | 	at org.springframework.cloud.bootstrap.config.PropertySourceBootstrapConfiguration.initialize(PropertySourceBootstrapConfiguration.java:101)
jhipster-registry_1   | 	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:626)
jhipster-registry_1   | 	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:370)
jhipster-registry_1   | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
jhipster-registry_1   | 	at io.github.jhipster.registry.JHipsterRegistryApp.main(JHipsterRegistryApp.java:71)
jhipster-registry_1   | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
jhipster-registry_1   | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
jhipster-registry_1   | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
jhipster-registry_1   | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
jhipster-registry_1   | 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
jhipster-registry_1   | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
jhipster-registry_1   | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:51)
jhipster-registry_1   | 	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:52)

```

**Related issues**

The issue has been reported in spring-cloud-config and it is supposed to be fixed, but the intended setup described in #1485 is not working.
[Config Server using GitHub AND Vault AND bootstrap = true not working #1485](https://github.com/spring-cloud/spring-cloud-config/issues/1485)
[Made sure configTokenProvider default bean will be overridden by more specific versions. Fixing Issue #1485 #1537](https://github.com/spring-cloud/spring-cloud-config/pull/1537)

**JHipster Registry Version(s)**

6.1.2

### PR DESCRIPTION
**Pull request by vishal423**
Closes #433 
--
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/jhipster-registry/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-registry/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,3 +27,78 @@ To run the cloned repository;
 [github-actions-build]: https://github.com/jhipster/jhipster-registry/workflows/Build/badge.svg
 [github-actions-docker]: https://github.com/jhipster/jhipster-registry/workflows/Docker%20Image/badge.svg
 [github-actions-url]: https://github.com/jhipster/jhipster-registry/actions
+
+## Vault Integration
```
Comment (gmarziou) [inline README.md:31]: Maybe refer to "Hashicorp Vault" as there are several products named "Vault".

Comment (vishal423) [reply README.md:31]: sure. Anything else @gmarziou you would like to change?

Comment (gmarziou) [reply README.md:31]: No, it's great

**Comment by gmarziou** [comment]
Great work

**Comment by vishal423** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/42768

**Comment by pascalgrimaud** [comment]
@vishal423 : approved

**Comment by pascalgrimaud** [comment]
@vishal423 @gmarziou : do you think we should add some documation at jhipster.tech too ?

**Comment by gmarziou** [comment]
Maybe adding a short mention in https://www.jhipster.tech/jhipster-registry/#security with a link to https://github.com/jhipster/jhipster-registry#readme

**Comment by mraible** [comment]
I'd like to do a blog post on the @oktadev blog too. If any of y'all want to write it, I can help make it happen!

**Pull request merged**